pkg/collector/device.go

```diff
@@ -44,67 +44,59 @@ func (c *deviceCollector) Collect(ch chan<- prometheus.Metric) {
 		if item.NeedUpgrade {
 			needUpgrade = 1
 		}
-		ch <- prometheus.MustNewConstMetric(c.omadaDeviceUptimeSeconds, prometheus.GaugeValue, item.Uptime,
-			item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type)
-
-		ch <- prometheus.MustNewConstMetric(c.omadaDeviceCpuPercentage, prometheus.GaugeValue, item.CpuUtil,
-			item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type)
-
-		ch <- prometheus.MustNewConstMetric(c.omadaDeviceMemPercentage, prometheus.GaugeValue, item.MemUtil,
-			item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type)
-
-		ch <- prometheus.MustNewConstMetric(c.omadaDeviceNeedUpgrade, prometheus.GaugeValue, needUpgrade,
-			item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type)
+		labels := []string{item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type}
 
+		ch <- prometheus.MustNewConstMetric(c.omadaDeviceUptimeSeconds, prometheus.GaugeValue, item.Uptime, labels...)
+		ch <- prometheus.MustNewConstMetric(c.omadaDeviceCpuPercentage, prometheus.GaugeValue, item.CpuUtil, labels...)
+		ch <- prometheus.MustNewConstMetric(c.omadaDeviceMemPercentage, prometheus.GaugeValue, item.MemUtil, labels...)
+		ch <- prometheus.MustNewConstMetric(c.omadaDeviceNeedUpgrade, prometheus.GaugeValue, needUpgrade, labels...)
 		if item.Type == "ap" {
-			ch <- prometheus.MustNewConstMetric(c.omadaDeviceTxRate, prometheus.GaugeValue, item.TxRate,
-				item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type)
-
-			ch <- prometheus.MustNewConstMetric(c.omadaDeviceRxRate, prometheus.GaugeValue, item.RxRate,
-				item.Name, item.Model, item.Version, item.Ip, item.Mac, site, client.SiteId, item.Type)
+			ch <- prometheus.MustNewConstMetric(c.omadaDeviceTxRate, prometheus.GaugeValue, item.TxRate, labels...)
```

Comment:
Would this be a Counter instead of a Gauge? It looks like the `rx`/`tx` returned from the API is a total value and not the current value

Reply:
Yes, updated.